internal/resources/resource_lb_monitors.go

```diff
@@ -25,91 +32,25 @@ func LoadBalancerMonitors() *schema.Resource {
 				Description: "Creating the Network Load balancer Monitor.",
 				Optional:    true,
 			},
-			"send_type": {
-				Type:        schema.TypeString,
-				Description: "send type method like GET,POST",
-				Optional:    true,
-			},
-			"monitor_timeout": {
-				Type:        schema.TypeInt,
-				Optional:    true,
-				Description: "Timeout for Network loadbalancer Monitor",
-			},
-			"monitor_interval": {
-				Type:        schema.TypeInt,
-				Description: "Interval time for Network loadbalancer Monitor",
-				Optional:    true,
-			},
-			"send_version": {
-				Type:        schema.TypeString,
-				Description: "Network loadbalancer Monitor http version",
-				Optional:    true,
-			},
-			"send_data": {
-				Type:        schema.TypeString,
-				Description: "Network loadbalancer Monitor Send info",
-				Optional:    true,
-			},
-			"receive_data": {
-				Type:        schema.TypeString,
-				Description: "Network loadbalancer Monitor receive info",
-				Optional:    true,
-			},
-			"receive_code": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Description: "Network loadbalancer Monitor receive status codes like 200,300,301,302,304,307",
-			},
-			"monitor_destination": {
-				Type:        schema.TypeString,
-				Optional:    true,
-				Description: "Network loadbalancer Monitor destination",
-			},
-			"monitor_reverse": {
-				Type:        schema.TypeBool,
-				Default:     true,
-				Optional:    true,
-				Description: "Network loadbalancer Monitor Reverse",
-			},
-			"monitor_transparent": {
-				Type:        schema.TypeBool,
-				Default:     true,
-				Optional:    true,
-				Description: "Network loadbalancer Monitor transparent",
-			},
-			"monitor_adaptive": {
-				Type:        schema.TypeBool,
-				Optional:    true,
-				Description: "Network loadbalancer Monitor adaptive",
-			},
-			"fall_count": {
-				Type:        schema.TypeInt,
-				Optional:    true,
-				Description: "Network loadbalancer Monitor fall counts",
-			},
-			"rise_count": {
-				Type:        schema.TypeInt,
-				Optional:    true,
-				Description: "Network loadbalancer Monitor rise counts",
-			},
-			"alias_port": {
-				Type:        schema.TypeInt,
-				Optional:    true,
-				Description: "Network loadbalancer Monitor alias port",
-			},
-			"monitor_type": {
+			"type": {
 				Type:             schema.TypeString,
 				ValidateDiagFunc: validations.StringInSlice([]string{"LBHttpMonitorProfile", "LBHttpsMonitorProfile", "LBIcmpMonitorProfile", "LBPassiveMonitorProfile", "LBTcpMonitorProfile", "LBUdpMonitorProfile"}, false),
 				Required:         true,
 				Description:      "Network Loadbalancer Supported values are `LBHttpMonitorProfile`,`LBHttpsMonitorProfile`, `LBIcmpMonitorProfile`, `LBPassiveMonitorProfile`,`LBTcpMonitorProfile`, `LBUdpMonitorProfile`",
 			},
+			"http_monitor":    schemas.HttpMonitorSchema(),
```

Comment:
are u making sure we are handling the scenario where user specifies a type field and has defined the respective fields only ? are we erroring out ? also are we making sure only one can be configured at a time ? i mean user should not mention http and https at a give point of time in the same resource definition

Reply:
updated

Reply:
still needs change